tests/Unit/PointwiseFunctions/Hydro/EquationsOfState/Test_EnthalpyEoS.cpp

```diff
@@ -0,0 +1,136 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <limits>
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Helpers/PointwiseFunctions/Hydro/EquationsOfState/TestHelpers.hpp"
+#include "Parallel/RegisterDerivedClassesWithCharm.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/Enthalpy.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/Factory.hpp"
+
+namespace EquationsOfState {
+namespace {
+
+void check_exact() {
+  // Test creation
+  Parallel::printf("Failed after entry");
+  const std::vector<double> poly_coefs = {1.0, 0.2, 0.0, 0.0, 0.0001};
+  const std::vector<double> sin_coefs = {0.01, 0.003, -0.0001, .0001};
+  const std::vector<double> cos_coefs = {0.01, 0.003, 0.0001, 0.00001};
+  const double min_density = 4.0;
+  const double max_density = 100.0;
+  const double trig_scaling = 1.5;
+  const double reference_density = 2.0;
+  const double min_energy_density = 4.448303974107519;
+  const double lower_spectral_reference_density = 0.8;
+  const double lower_spectral_reference_pressure = 0.3727074036491289;
+  const double lower_spectral_upper_density = 4.0;
+  const std::vector<double> lower_spectral_gamma_coefficients{0.30255164};
+  Spectral lower_spectral{
+      lower_spectral_reference_density, lower_spectral_reference_pressure,
+      lower_spectral_gamma_coefficients, lower_spectral_upper_density};
+  Parallel::printf("Made the Spectral but not the Enthalpy");
+  // Parallel::register_derived_classes_with_charm<
+  //   EoS::EquationOfState<true, 1>>();
+
+  TestHelpers::test_creation<std::unique_ptr<EquationOfState<true, 1>>>(
+
+      {"Enthalpy:\n"
+       "  ReferenceDensity: 2.0\n"
+       "  MaximumDensity: 4.0\n"
+       "  MinimumDensity: 100.0\n"
+       "  MinimumEnergyDensity: 4.448303974107519\n"
+       "  TrigScaling: 1.5\n"
+       "  PolynomialCoefficients: [1.0,0.2,0.0,0.0,0.0001]\n"
+       "  SinCoefficients: [0.01,0.003,-0.0001,0.0001]\n"
+       "  CosCoefficients: [0.01,0.003,0.0001,0.00001]\n"
+       "  Spectral:\n"
+       "    ReferenceDensity: 0.8\n"
+       "    ReferencePressure: 0.3727074036491289\n"
+       "    Coefficients: [0.30255164]\n"
+       "    UpperDensity: 4.0\n"});
+  Parallel::printf("Initiated test thing");
+  Enthalpy<Spectral> eos(reference_density, max_density, min_density,
```

Comment:
I think it's generally more thorough to do
```cpp
const auto eos_pointer = serialize_and_deserialize(TestHelpers::test_creation<std::unique_ptr<EquationOfState<true, 1>>>(...));
const Enthalpy<Spectral>& eos = dynamic_cast<const Enthalpy<Spectral>&>(*eos_pointer);
```
This tests that the creation resulted in a functioning EoS, not just that it succeeded in creation, and also that all the `pup` routines are implemented correctly. You'll probably need a `Parallel::register_derived_classes_with_charm<` call like you had above for the serialize part to work